synapse/federation/federation_client.py

```diff
@@ -643,6 +649,14 @@ async def _check_sigs_and_hash_and_fetch_one(
                 pdu.event_id,
                 e,
             )
+            log_kv(
+                {
+                    "message": "Signature on retrieved event was invalid. "
+                    "Checking local store/orgin server",
```

Comment:
Why do we `get_pdu` from a remote origin homeserver again?

The events we pass to `_check_sigs_and_hash_and_fetch` were already fetched and that's why we're checking them here.

Reply:
From the [comment doc](https://github.com/matrix-org/synapse/blob/3dd175b628bab5638165f20de9eade36a4e88147/synapse/federation/federation_client.py#L570-L571), it looks like we ask the original server that sent the message which could be different than how we encountered it initially.

> request it from the sender's server (if that is different from `origin`)